src/api_server/src/attest/amber.rs

```diff
@@ -0,0 +1,141 @@
+}
+
+pub struct Amber {
+    config: AmberConfig,
+    certs: jwk::JwkSet,
+}
+
+#[async_trait]
+impl Attest for Amber {
+    async fn attest_verify(
+        &mut self,
+        tee: Tee,
+        _nonce: &str,
+        attestation: &str,
+    ) -> Result<AttestationResults> {
+        if tee != Tee::Tdx {
+            bail!("Only implement for tdx now");
+        }
+        // get quote
+        let attestation = serde_json::from_str::<Attestation>(attestation)
+            .map_err(|e| anyhow!("Deserialize Attestation failed: {:?}", e))?;
+        let tdx_evidence = serde_json::from_str::<TdxEvidence>(&attestation.tee_evidence)
+            .map_err(|e| anyhow!("Deserialize TDX Evidence failed: {:?}", e))?;
+
+        // construct attest request data
+        let req_data = AttestReqData {
+            quote: tdx_evidence.quote,
+        };
+
+        let attest_req_body = serde_json::to_string(&req_data)
+            .map_err(|e| anyhow!("Serialize attestation request body failed: {:?}", e))?;
+
+        // send attest request
+        log::info!("post attestation request ...");
+        let client = reqwest::Client::new();
+        let resp = client
+            .post(format!("{}/appraisal/v1/attest", &self.config.base_url))
+            .header(CONTENT_TYPE, "application/json")
+            .header(ACCEPT, "application/json")
+            .header("x-api-key", &self.config.api_key)
+            .body(attest_req_body)
+            .send()
+            .await
+            .map_err(|e| anyhow!("Post attestation request failed: {:?}", e))?;
+
+        if resp.status() != reqwest::StatusCode::OK {
+            bail!(
+                "Attestation request failed: respone status={}",
+                resp.status()
+            );
+        }
+
+        // get token kid
+        let resp_data = resp
+            .json::<AttestRespData>()
+            .await
+            .map_err(|e| anyhow!("Deserialize attestation respone failed: {:?}", e))?;
+        let header = decode_header(&resp_data.token)
+            .map_err(|e| anyhow!("Decode token header failed: {:?}", e))?;
+        let kid = header.kid.ok_or(anyhow!("Token missing kid"))?;
+
+        log::debug!("token={}", &resp_data.token);
+
+        // find jwk
+        let key = self.certs.find(&kid).ok_or(anyhow!("Find jwk failed"))?;
+        let alg = key.common.algorithm.ok_or(anyhow!("Get jwk alg failed"))?;
+
+        // verify and decode token
+        let dkey = DecodingKey::from_jwk(&key)?;
+        let token = decode::<Claims>(&resp_data.token, &dkey, &Validation::new(alg))
+            .map_err(|e| anyhow!("Decode token failed: {:?}", e))?;
+
+        // check unmatched policy
+        let allow = self.config.allow_unmatched_policy.unwrap_or(false);
+        if allow == false && token.claims.amber_unmatched_policy_ids.is_some() {
+            bail!("Evidence doesn't match policy");
+        }
+
+        let result = AttestationResults::new(tee, true, None, None, None);
```

Comment:
Can we return some more information in the attestation result instead of None? For example, the key content of the parsed evidence, as well as the status of some TCBs, etc.

Reply:
Amber Attestation response is a JWT. In addition to standard claims, the JWT include many Amber defined claims such as 


amber_tdx_seamsvn, 
amber_tdx_rtmr0, 
amber_tdx_rtmr1, 
amber_tdx_rtmr2,
amber_tdx_rtmr3,
amber_tdx_mrtd,
amber_tdx_mrsignerseam,
amber_tdx_mrseam,
amber_matched_policy_ids, 
amber_unmatched_policy_ids,
amber_custom_policy,
….


It is different from AttestationResults in design, so I prefer to define an interface to cover different AS attestation results in high level, and analyze or decode the result by AttestVerifier itself. The benefit is there is no need to care about the detail of AS attestation results in high level, and AttestVerifier no need to translate the external AS attestation result to COCO AS AttestationResults.